tests/test_distributions.py

```diff
@@ -1,490 +1,12 @@
-        },
-    },
-}
-
-
-unsupported_dtype_distributions = [
-    "Poisson",
-    "NegativeBinomial",
-    "Multinomial",
-    "Geometric",
-    "DiscreteUniform",
-    "Categorical",
-    "BetaBinomial",
-    "Binomial",
-]
-
-_distribution_extra_parameters = {}
-for distribution in _distribution_conditions:
-    extra_parameters = {
-        "dtype": None,
-        "validate_args": {"validate_args": True},
-        "allow_nan_stats": {"allow_nan_stats": True},
-    }
-    if (
-        not issubclass(getattr(pm, distribution, None), pm.ContinuousDistribution)
-        and distribution not in unsupported_dtype_distributions
-    ):
-        extra_parameters["dtype"] = {"dtype": "int32"}
-    if distribution == "AR":
-        # time series cannot be configured at the moment
-        extra_parameters = {
-            "dtype": None,
-            "validate_args": None,
-            "allow_nan_stats": None,
-        }
-    _distribution_extra_parameters[distribution] = extra_parameters
-
-
-@pytest.fixture(scope="function", params=list(_distribution_conditions), ids=str)
-def distribution(request):
-    return request.param
-
-
-@pytest.fixture(
-    scope="function", params=["scalar_parameters", "multidim_parameters"], ids=str,
-)
-def distribution_conditions(distribution, request):
-    conditions = _distribution_conditions[distribution][request.param]
-    if conditions is None:
-        pytest.skip("Distribution does not support {}".format(request.param))
-    else:
-        conditions = conditions.copy()
-    log_prob_test_sample = conditions.pop("sample", 0.1)
-    expected_log_prob = conditions.pop("expected", None)
-    return distribution, conditions, log_prob_test_sample, expected_log_prob
-
-
-@pytest.fixture(scope="function", params=["dtype", "validate_args", "allow_nan_stats"], ids=str)
-def distribution_extra_parameters(distribution, request):
-    conditions = _distribution_conditions[distribution]["scalar_parameters"]
-    return (
-        distribution,
-        request.param,
-        conditions,
-        _distribution_extra_parameters[distribution][request.param],
-    )
-
-
-@pytest.fixture(scope="function", params=list(_check_broadcast), ids=str)
-def broadcast_distribution(request):
-    return request.param
-
-
-@pytest.fixture(scope="function", ids=str)
-def check_broadcast(broadcast_distribution, request):
-    conditions = _check_broadcast[broadcast_distribution]
-    batch_stack = conditions.pop("batch_stack", (1, 2))
-    event_stack = conditions.pop("event_stack", (3, 4))
-    samples = conditions.pop("samples")
-    dist_class = getattr(pm, broadcast_distribution)
-    dist = dist_class(name=broadcast_distribution, batch_stack=batch_stack, event_stack=event_stack)
-    return dist, samples
-
+from .fixtures.fixtures_distributions import distribution, distribution_conditions, distribution_extra_parameters, broadcast_distribution, check_broadcast
```

Comment:
Request for help - what is the "right" way of importing lots of these? I tried * but that came with a depreciation warning.

Reply:
Since we have a directory scope for `conftest.py`. Can we just create a directory hierarchy for the test functions, i.e. separate directories for sampling, execution, gp tests. Each directory will have its own `conftest.py` and the python file (`fixtures.py` for example) of the fixtures. Then to import all the fixtures from `fixtures.py` we can just load a plugin (which in our case is just a python file `fixtures.py`). Something like this:

```python
# our tests/sampling/conftest.py
pytest_plugins = [                                                        
     "fixtures", # fixtures are in tests/sampling/fixtures.py
]
```
Directory scoping also will help us to avoid any ambiguity in fixtures we could have in future. Plus depending on the directory we can have separate logic for hooks.